bin/generate_practice_exercise

```diff
@@ -19,28 +41,60 @@ if ! [ -x "$(command -v emacs)" ]; then
     exit 1
 fi
 
-SLUG="$1"
-exercise_dir="exercises/practice/${SLUG}"
+require_files_template "solution"
+require_files_template "test"
+require_files_template "example"
 
-download() {
-    file="$1"
-    url="$2"
-    curl --silent --show-error --fail --retry 3 --max-time 3 \
-        --output "$file" "$url"
-}
+[[ -f ./bin/fetch-configlet ]] || die "Run this script from the repo's root directory."
+
+author=''
+difficulty=''
+while getopts :ha:d: opt; do
+    case $opt in
+        h) help_and_exit ;;
+        a) author=$OPTARG ;;
+        d) difficulty=$OPTARG ;;
+        ?) echo >&2 "Unknown option: -$OPTARG"; help_and_exit ;;
+    esac
+done
+shift "$((OPTIND - 1))"
+
+(( $# >= 1 )) || help_and_exit
+
+slug="${1}"
+
+if [[ -z "${author}" ]]; then
+    read -rp 'Your GitHub username: ' author
+fi
+
+if [[ -z "${difficulty}" ]]; then
+    read -rp "What's the difficulty for ${slug}? " difficulty
+fi
 
-# build configlet
 echo "Fetching latest version of configlet..."
 ./bin/fetch-configlet
 
-# Preparing config.json
 echo "Adding instructions and configuration files..."
-./bin/configlet create --practice-exercise "$SLUG"
+./bin/configlet create --practice-exercise "${slug}" --author "${author}" --difficulty "${difficulty}"
+
+filter='.exercises.practice = (.exercises.practice | sort_by(.difficulty, .slug))'
+jq "${filter}" config.json > config.sorted && mv config.sorted config.json
 
 pushd tools
-emacs -batch -l install-packages.el -l practice-exercise-generator.el --eval "(exercism/generate-practice-exercise \"$SLUG\")"
+emacs -batch -l install-packages.el -l practice-exercise-generator.el --eval "(exercism/generate-practice-exercise \"${slug}\")"
 popd
 
-echo "All stub files were created. After implementing the solution, tests and configuration, please run:"
-echo "    ./bin/configlet sync --update --tests --exercise ${SLUG}"
-echo "    ./bin/configlet fmt --update --yes --exercise ${SLUG}"
+exercise_dir="exercises/practice/${slug}"
+files=$(jq -r --arg dir "${exercise_dir}" '.files | to_entries | map({key: .key, value: (.value | map("'"'"'" + $dir + "/" + . + "'"'"'") | join(" and "))}) | from_entries' "${exercise_dir}/.meta/config.json")
+
+cat << NEXT_STEPS
+
+Your next steps are:
+- Implement the test suite in $(jq -r '.test' <<< "${files}")
```

Comment:
Maybe add a line like 'the test specifications have been copied into the test methods stubs by the generator' or smth?